.eslintrc.json

```diff
@@ -16,5 +16,23 @@
   "rules": {
     "padding-line-between-statements": "off",
     "react/jsx-uses-vars": "error"
-  }
+  },
+  "overrides": [
+    {
+      "files": ["*.ts", "*.tsx"],
+      "parser": "@typescript-eslint/parser",
+      "parserOptions": {
+        "project": "./tsconfig.json"
+      },
+      "extends": [
+        "plugin:@typescript-eslint/eslint-recommended",
+        "plugin:@typescript-eslint/recommended",
+        "plugin:@typescript-eslint/recommended-requiring-type-checking"
+      ],
+      "rules": {
+        "@typescript-eslint/explicit-function-return-type": "off", // inference is usually useful
```

Comment:
@calebeby, does this mean you never need to specify the expected return type from functions?

Reply:
It means that ESLint won't yell at you telling you that you have to. If this rule was enabled, it would require you to specify a return type for every function.

TS type inference is really helpful. I don't think it is useful to specify a return type from every function. Most of the time, TS knows the return type without you telling it. In some cases it is necessary to specify a return type (recursive function that returns a call to itself, you must specify a return type, but TS will tell you that, ESLint doesn't need to). Other times it is helpful to but not required (like when you have multiple branches that each have their own returns, or if you are returning an object that you want to be validated within the function rather than wherever the function is called).

Reply:
I amended my comment before refreshing this to see your response. That's helpful. At some point, somewhere, this should be in a guide.